src/common/cookies.js

```diff
@@ -0,0 +1,53 @@
+/**
+  @type {CookieConsent}
+ */
+export const defaultConsent = {
+  analytics: null
+}
+
+/**
+ * Parses the cookie consent policy
+ * @param {string} value
+ */
+export function parseCookieConsent(value) {
+  /** @type {CookieConsent} */
+  let cookieConsent
+
+  try {
+    const encodedValue = decodeURIComponent(value)
+
+    // eslint-disable-next-line @typescript-eslint/no-unsafe-assignment
+    const decodedValue = JSON.parse(encodedValue)
+
+    if (isValidConsent(decodedValue)) {
+      cookieConsent = decodedValue
+    } else {
+      cookieConsent = defaultConsent
+    }
+  } catch {
+    cookieConsent = defaultConsent
+  }
+
+  return cookieConsent
+}
+
+/**
+ * Serialises the cookie consent policy
+ * @param {CookieConsent} consent
+ * @returns {string} cookie value
+ */
+export function serialiseCookieConsent(consent) {
+  return encodeURIComponent(JSON.stringify(consent))
+}
+
+/**
+ * @param {any} consent
```

Comment:
Try changing this to `unknown` and we've got a bit of work to do

The `in consent` bit will throw a TypeError unless we check it's a non-null, non-array object first

```suggestion
 * @param {unknown} consent
```

<br>

> #### [`JSON.parse()` return value](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/JSON/parse#return_value)
> The [Object](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Object), [Array](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Global_Objects/Array), string, number, boolean, or null value corresponding to the given JSON text.